x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/action/EsqlQueryResponseTests.java

```diff
@@ -125,7 +125,10 @@ EsqlQueryResponse randomResponseAsync(boolean columnar, EsqlQueryResponse.Profil
 
     private ColumnInfo randomColumnInfo() {
         DataType type = randomValueOtherThanMany(
-            t -> false == DataTypes.isPrimitive(t) || t == EsqlDataTypes.DATE_PERIOD || t == EsqlDataTypes.TIME_DURATION,
+            t -> false == DataTypes.isPrimitive(t)
+                || t == EsqlDataTypes.DATE_PERIOD
+                || t == EsqlDataTypes.TIME_DURATION
+                || EsqlDataTypes.isCounterType(t),
             () -> randomFrom(EsqlDataTypes.types())
```

Comment:
I always get this backwards - can the `type` include a counter type here? I think it should because we can return them.

Reply:
I removed this restriction in https://github.com/elastic/elasticsearch/pull/107877/commits/5c2b08fbbb180065b34ae4ea50cb377fafe16712